src/HtmlGenerator.Tests/GitBlameVisitorTests.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using GitGlyph;
+using LibGit2Sharp;
+using Microsoft.SourceBrowser.HtmlGenerator.Utilities;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace HtmlGenerator.Tests
+{
+    [TestClass]
+    public class GitBlameVistorTests
+    {
+        [TestMethod]
+        public void TestVisit()
+        {
+            string pathToTempDirectory = Path.GetTempPath() + "GitBlameVisitorRepo\\";
+            string pathToTestFile = pathToTempDirectory + "Test.cs";
+            string pathToGitBlameRepo = Directory.GetCurrentDirectory() + "\\Repositories\\GitBlameVisitorRepo.git";
+            Repository.Clone(pathToGitBlameRepo, pathToTempDirectory);
+            try
+            {
+                Repository repo = new Repository(pathToTempDirectory + ".git");
+                //Test.cs contains:
+                //bd46cc8 public class Test
+                //bd46cc8 {
+                //909a6197  private int test;
+                //11933d2a  private int myInt;
+                //bd46cc8 }
+                Dictionary<string, string> context = new Dictionary<string, string>
+                {
+                    {Microsoft.SourceBrowser.MEF.ContextKeys.FilePath, pathToTestFile},
+                    {Microsoft.SourceBrowser.MEF.ContextKeys.LineNumber, "3"}
+                };
+                string expectedThirdLineSha1 = "909a6197";
+                GitBlameVisitor visitor = new GitBlameVisitor(repo, new PluginLogger());
+
+                string htmlResult = visitor.Visit(null, context);
+
+                Assert.IsTrue(htmlResult.Contains(expectedThirdLineSha1));
+
+            }
+            finally
+            {
+                //Remove old repository which contains the Test.cs file
+                DeleteDirectory(pathToTempDirectory);
+            }
+
+        }
+
+
+        /// <summary>
+        /// Recursively unsets the readonly bit for all filesand repositories in a directory and delete this directory

```

Comment:
can you state it is "necessary for Git repositories" if this the reason why you introduce this method?